### DISCUSSION TITLE
Visual Studio and Github Copilot chat

### DISCUSSION THREAD

**Discussion by willhowell18** [community/community#78128] [Copilot Conversations] [answered]
### Select Topic Area

Question

### Body

Hi

I have installed Github Copilot and Github Copilot chat extensions into visual studio 2022
Github Copilot appears to be working 
However Github Copilot chat is greyed out on the "view" menu and in the output I get the error message below: 

Logging to: C:\Users\whowell\AppData\Local\Temp\VSGitHubCopilotLogs\20231207_094027.541_VSGitHubCopilot.chat.log
[Conversations Information] Copilot chat version 0.1.1982-beta+1e5d439ddf (0.1.1982.7773). VS: VisualStudio.17.Release/17.6.2+33723.286. Session: a5d7dda5-81f6-4e55-9af0-54536e8e8d4e. 
[ConversationsServiceManager Information] Out-of-process service finished initializing.
[Conversations Information] [CopilotClient] Need to get an auth token.
[Conversations Information] [CopilotClient] Need to get an auth token.
[Conversations Error] [CopilotClient] Copilot token request failed: System.NullReferenceException: Object reference not set to an instance of an object.
   at Microsoft.VisualStudio.Conversations.Service.CopilotRetryPolicy.OnRetryLogFirstRetries(DelegateResult`1 response, TimeSpan waitDuration) in D:\a\_work\1\s\src\Conversations.Service.Implementation\CopilotRetryPolicy.cs:line 58
   at Microsoft.VisualStudio.Conversations.Service.CopilotRetryPolicy.<CreatePolicy>b__5_2(DelegateResult`1 response, TimeSpan waitDuration) in D:\a\_work\1\s\src\Conversations.Service.Implementation\CopilotRetryPolicy.cs:line 43
   at Polly.AsyncRetryTResultSyntax.<>c__DisplayClass20_0`1.<<WaitAndRetryAsync>b__0>d.MoveNext()
--- End of stack trace from previous location ---
   at Polly.Retry.AsyncRetryEngine.ImplementationAsync[TResult](Func`3 action, Context context, CancellationToken cancellationToken, ExceptionPredicates shouldRetryExceptionPredicates, ResultPredicates`1 shouldRetryResultPredicates, Func`5 onRetryAsync, Int32 permittedRetryCount, IEnumerable`1 sleepDurationsEnumerable, Func`4 sleepDurationProvider, Boolean continueOnCapturedContext)
   at Polly.AsyncPolicy`1.ExecuteAsync(Func`3 action, Context context, CancellationToken cancellationToken, Boolean continueOnCapturedContext)
   at Microsoft.Extensions.Http.PolicyHttpMessageHandler.SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
   at Microsoft.Extensions.Http.Logging.LoggingScopeHttpMessageHandler.<SendAsync>g__Core|5_0(HttpRequestMessage request, CancellationToken cancellationToken)
   at System.Net.Http.HttpClient.<SendAsync>g__Core|83_0(HttpRequestMessage request, HttpCompletionOption completionOption, CancellationTokenSource cts, Boolean disposeCts, CancellationTokenSource pendingRequestsCts, CancellationToken originalCancellationToken)
   at Microsoft.VisualStudio.Conversations.CopilotClient.CopilotAuthTokenManager.GetCopilotIdentityAsync(CancellationToken cancellationToken, Nullable`1 correlationId) in D:\a\_work\1\s\src\CopilotClient\CopilotAuthTokenManager.cs:line 192
[Conversations Error] [CopilotClient] Copilot token request failed: System.NullReferenceException: Object reference not set to an instance of an object.
   at Microsoft.VisualStudio.Conversations.Service.CopilotRetryPolicy.OnRetryLogFirstRetries(DelegateResult`1 response, TimeSpan waitDuration) in D:\a\_work\1\s\src\Conversations.Service.Implementation\CopilotRetryPolicy.cs:line 58
   at Microsoft.VisualStudio.Conversations.Service.CopilotRetryPolicy.<CreatePolicy>b__5_2(DelegateResult`1 response, TimeSpan waitDuration) in D:\a\_work\1\s\src\Conversations.Service.Implementation\CopilotRetryPolicy.cs:line 43
   at Polly.AsyncRetryTResultSyntax.<>c__DisplayClass20_0`1.<<WaitAndRetryAsync>b__0>d.MoveNext()
--- End of stack trace from previous location ---
   at Polly.Retry.AsyncRetryEngine.ImplementationAsync[TResult](Func`3 action, Context context, CancellationToken cancellationToken, ExceptionPredicates shouldRetryExceptionPredicates, ResultPredicates`1 shouldRetryResultPredicates, Func`5 onRetryAsync, Int32 permittedRetryCount, IEnumerable`1 sleepDurationsEnumerable, Func`4 sleepDurationProvider, Boolean continueOnCapturedContext)
   at Polly.AsyncPolicy`1.ExecuteAsync(Func`3 action, Context context, CancellationToken cancellationToken, Boolean continueOnCapturedContext)
   at Microsoft.Extensions.Http.PolicyHttpMessageHandler.SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
   at Microsoft.Extensions.Http.Logging.LoggingScopeHttpMessageHandler.<SendAsync>g__Core|5_0(HttpRequestMessage request, CancellationToken cancellationToken)
   at System.Net.Http.HttpClient.<SendAsync>g__Core|83_0(HttpRequestMessage request, HttpCompletionOption completionOption, CancellationTokenSource cts, Boolean disposeCts, CancellationTokenSource pendingRequestsCts, CancellationToken originalCancellationToken)
   at Microsoft.VisualStudio.Conversations.CopilotClient.CopilotAuthTokenManager.GetCopilotIdentityAsync(CancellationToken cancellationToken, Nullable`1 correlationId) in D:\a\_work\1\s\src\CopilotClient\CopilotAuthTokenManager.cs:line 192

**Comment by rheapatel23** [comment]
Hi @willhowell18, Thanks for bringing this to our attention our team will look into what is happening here. Just an additional question: Is chat working when you search for it or not at all? 

Let me know, Rhea 
Product manager for GitHub Copilot in Visual Studio

**Reply by willhowell18** [reply]
Sorry just saw your message.
Not sure what you mean by "is chat working when you search for it or not at all?" How do you search for it in visual studio?

GitHub Copilot appears to be working. Its offering me code suggestions as I type and If I look at the visual studio "output" window and show output from "GitHub Copilot" there is lots of output and it seems ok 

However the "output" window for "GitHub Copilot chat" shows the error (fully printed in first post):  
"[Conversations Error] [CopilotClient] Copilot token request failed: System.NullReferenceException: Object 
reference not set to an instance of an object."
Also in the Visual studio "View" Menu GitHub Copilot chat is listed but is greyed out.


I am now paying for: 
GitHub Pro $4.00/month 
GitHub Copilot  $10.00/month

**Reply by willhowell18** [reply]
On my local PC at home: 
- GitHub copilot Works
- GitHub copilot chat works 

On my development PC at work (where I actually want to use it) : 
- GitHub copilot  Works 
- GitHub copilot chat **Does Not work** 

Both use the same version of Visual Studio 2022 and the same GitHub account

I suspect its a network setting preventing GitHub Copilot Chat communicating with the outside world. However I'm not a network expert so I have no idea what setting needs to change.

**Reply by CarlosDMSantos** [reply]
Hi!

Any news on this? We're experiencing the exact same issue. At my home both extensions work on Visual Studio, but at work the Copilot Chat extension for Visual Studio throws the reported NullReferenceException. I'm evaluating Github Copilot to decide if we shall subscribe to the service at work, so it's important that we have access to Copilot Chat in Visual Studio. The Copilot Chat extension for Visual Studio Code works fine at work, only the extension for Visual Studio has this problem.

Thank you,

Carlos.

**Reply by ayim** [reply]
@willhowell18 hang tight - can ya confirm your extension version?
I'd also ask that you try updating to the latest version if you're still running into that Null Reference Exception ... this was fixed.

@willhowell18  @CarlosDMSantos pinged the dev team on network/proxy troubleshooting.
It should be reading your system proxy settings but our endpoints sometimes aren't whitelisted on corporate proxy/firewall.
Seeing if we have a list or further troubleshooting steps for ya.

--
Aaron Yim
Product Manager for GitHub Copilot in Visual Studio.

**Reply by ayim** [reply]
@willhowell18 @CarlosDMSantos ping on this^

**Reply by willhowell18** [reply]
co-pilot is 1.155.0.0
I tried uninstalling co-pilot chat to reinstall it. However, It is no longer listed under online extensions 
Tried to download it manually (using link below) I get an error when I try and install the VSIX "This extension is not installable on any currently installed products"
https://marketplace.visualstudio.com/items?itemName=VisualStudioExptTeam.VSGitHubCopilot
I'm seriously considering giving up and cancelling my subscription

**Comment by timenauters** [comment]
I have a GitHub Copilot $10.00/month and only see "Start 30-day Free Trial" in VS Code, Chat but im logged in...

**Reply by davecheney** [reply]
@timenauters i'm very sorry to hear that. please contact support, https://support.github.com/contact and they will be able to sort you out.

**Comment by bigswede74** [comment]
A lot of users are having issues at my company with GitHub Copilot Chat for VS2022 since early last week. 

1. The first issue was just an endless progress ring and GitHub Copilot Chat never loaded.
2. The second issue is GitHub Copilot Chat loads however when you ask a question you get an object reference exception.

I was able to resolve both issues by updating VS2022 to v17.8.6 and Un-installing/Re-installing GitHub Copilot Chat extension. I noticed that a third new extension was installed **GitHub Copilot Chat - Services**
![image](https://github.com/community/community/assets/23264080/b722d0de-c929-4b24-9a7b-70f153c916e0)

**Still Broken**
I however am the exception it seems, 4 other engineers performed the same steps but are still experiencing the second issue. They do not get the new **GitHub Copilot Chat - Services** extension shown above when Un-installing/Re-installing GitHub Copilot Chat extension. With the missing services the object reference error is still there. You cannot find that extension when searching for it online either.

Is there a place we can download the required **GitHub Copilot Chat - Services** extension explicitly?

**Reply by davecheney** [reply]
@bigswede74 i'm very sorry to hear this. The issue you are reporting is _not_ related to this change. Visual Studio Chat does not use the endpoint that was turned down yesterday.

I recommend:
a. Creating a new discussion post to keep the issues separate.
b. Contacting support https://support.github.com/contact who will be able to assist or escalate to the appropriate engineering team.

**Reply by bigswede74** [reply]
It seems some updates that were pushed with version GitHub Copilot Chat - v0.2.393.21236 have fixed the issues we were seeing. I will close this issue.

**Comment by bigswede74** [comment]
> @bigswede74 i'm very sorry to hear this. The issue you are reporting is _not_ related to this change. Visual Studio Chat does not use the endpoint that was turned down yesterday.
> 
> I recommend: a. Creating a new discussion post to keep the issues separate. b. Contacting support https://support.github.com/contact who will be able to assist or escalate to the appropriate engineering team.

Created a new discussion #103172

**Comment by willhowell18** [comment]
Still no luck getting co-pilot chat to work 

Looking at other people's posts I tried updating GitHub Co-pilot and uninstalling and reinstalling GitHub Co-pilot Chat. 
GitHub Co-pilot updated ok 
GitHub Co-pilot Chat Uninstalled ok  
however GitHub Co-pilot Chat is not listed in the downloadable extensions
Tried to download it manually (using link below) I get an error when I try and install the VSIX "This extension is not installable on any currently installed products" 
https://marketplace.visualstudio.com/items?itemName=VisualStudioExptTeam.VSGitHubCopilot
 I'm seriously considering giving up and cancelling my subscription

**Reply by rheapatel23** [reply]
Hey @willhowell18 
Please fill a bug here: https://learn.microsoft.com/en-us/visualstudio/ide/how-to-report-a-problem-with-visual-studio?view=vs-2022 

In addition, feel free to email me: rhea.patel@microsoft.com and I will try and help you solve this. 

Thanks, 
Rhea - Product Manager GitHub Copilot for Visual Studio

**Comment by willhowell18** [comment]
I got a little further...
I updated visual studio 2022 to the latest version
I was then able to install the GitHub Co-Pilot Chat Extension
It is no longer greyed out on the menu Under View -->  GitHub Co-Pilot Chat
:)
Still doesn't work though:  :(
Logging to: C:\Users\whowell\AppData\Local\Temp\VSGitHubCopilotLogs\20240206_144551.783_VSGitHubCopilot.chat.log
[Conversations Information] Copilot chat version 0.2.393-beta+52f40db97c.RR (0.2.393.21236). VS: VisualStudio.17.Release/17.8.6+34525.116. Session: 70d356ef-b73f-411d-a321-2c25063109ec. 
[ConversationsServiceManager Information] Out-of-process service finished initializing.
[Conversations Information] [CopilotClient] Need to get an auth token.
[Conversations Information] [CopilotClient] Need to get an auth token.

**Reply by ayim** [reply]
Ah yes, latest version of chat requires VS17.8 to work.
Have you signed in to GitHub via the auth panel in the top right?
![image](https://github.com/community/community/assets/10951869/7122649c-d5b2-4c25-b202-b1dd0fd104ed)

**Reply by ayim** [reply]
If you have signed in and are still experiencing issues, please use the "Report a Problem" feature here in the top right of Visual Studio so we can collect logs and take a closer look.

![image](https://github.com/community/community/assets/10951869/a6230fb4-5c25-45e3-a239-6bd2e0033946)

**Comment by willhowell18** [comment]
I have my account set up under accounts

![auth](https://github.com/community/community/assets/153005978/beaf63b2-7427-479b-b652-eb375802ddcf)

**Reply by ayim** [reply, marked as answer]
Thanks for getting back to us so quickly! Sorry we've had to go back and forth here.

If you have signed in and are still experiencing issues, please use the "Report a Problem" feature here in the top right of Visual Studio so we can collect logs and take a closer look.

![image](https://github.com/community/community/assets/10951869/7ec13465-3c7d-41a6-81a2-7bb0dd10718f)

**Comment by denissopovic** [comment]
I had this problem too in VS Code and I solved it by log out and log in on my account again in VS Code. It is probably some token that needed to get refreshed

<img width="504" alt="image" src="https://github.com/community/community/assets/10199560/f3cbe92a-a6ed-4701-8058-823b72144988">

**Comment by willhowell18** [comment]
I have tried: 
Installing the latest versions of everything again.
Removing and re-adding my GitHub account

The error has changed slightly I now get: a rotating wait icon on the GitHub Co-Pilot chat window and in the output I get the following message:

Logging to: C:\Users\whowell\AppData\Local\Temp\VSGitHubCopilotLogs\20240209_081650.138_VSGitHubCopilot.chat.log
[Conversations Information] Copilot chat version 0.2.393-beta+52f40db97c.RR (0.2.393.21236). VS: VisualStudio.17.Release/17.8.6+34525.116. Session: f0668cd1-ee98-4f3a-baf1-65d4d8307a8a. 
[ConversationsServiceManager Information] Out-of-process service finished initializing.
[Conversations Information] [CopilotClient] Need to get an auth token.
[Conversations Information] [CopilotClient] Need to get an auth token. 

I've tried raising this with the developer community so I'll see if they have an answer

**Reply by ayim** [reply]
devcom is definitely the right place to report this issue. If you have a feedback ticket # or link, please send this to me so that I can follow up more aggressively

**Comment by MishuBell** [comment]
I've been having the issue stated in the first post too. I believe it to be since i've accepted the update for VisualStudio Community ... As ayim said, if i understood correctly: GitHub Copilot Chat 'only' works with VisualStudio Community 17.8?

```
[Conversations Information] Copilot chat version 0.2.584-beta+b218379f84.RR (0.2.584.45592). VS: VisualStudio.17.Release/17.9.2+34622.214. Session: f9509ea6-74e3-42e3-8632-01956c06c5c9. 
[Conversations Information] Need to get an auth token.
[Conversations Error] Copilot token request failed: System.IO.FileNotFoundException: Die Datei oder Assembly "Polly, Version=7.0.0.0, Culture=neutral, PublicKeyToken=c8a3ffc3f8f825cc" oder eine Abhängigkeit davon wurde nicht gefunden. Das System kann die angegebene Datei nicht finden.
```

How do i get a new AuthToken? What is the fix to this? Do i need to go down from 17.9.2 to 17.8 again, if so - how?

**Reply by ayim** [reply]
17.8 and above should work; the behaviour you're experiencing doesn't seem right.
Would you mind using the "Report a Problem" tool so that we can take a deeper dive into telemetry and logs for you.
![image](https://github.com/community/community/assets/10951869/2c777a9c-062d-4932-8d99-a58fdb6e2a4c)

**Reply by MishuBell** [reply]
Used it. They've processed it and released a fix two days later. Awesome!

**Comment by Al-Nisour-University** [comment]
I have  issue I am experiencing with GitHub Copilot in Visual Studio Code. The tool was functioning properly, but it suddenly stopped working and the following problem occurred:
 when open copilot chat in visual studio code it appear 
there seems to be a problem with your account please contact github support
I have tried the following steps to resolve the issue, but to no avail:
1. Restarting Visual Studio Code.
2. Updating Visual Studio Code to the latest version.
3. Reinstalling the Copilot extension.
4. Ensuring a stable internet connection.
5. Logging out and logging back into my GitHub account.
